hibernate-core/src/test/java/org/hibernate/orm/test/query/resultmapping/NamedNativeQueryWithCountColumnTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.query.resultmapping;
+
+import org.hibernate.cfg.AvailableSettings;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.ColumnResult;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.NamedNativeQuery;
+import jakarta.persistence.SqlResultSetMapping;
+import jakarta.persistence.Table;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @author Nathan Xu
+ */
+@Jpa(
+		annotatedClasses = NamedNativeQueryWithCountColumnTest.Sample.class,
+		properties = @Setting(name = AvailableSettings.GLOBALLY_QUOTED_IDENTIFIERS, value = "true")
+)
+@TestForIssue(jiraKey = "HHH-15070")
+class NamedNativeQueryWithCountColumnTest {
+
+	@BeforeEach
+	public void setUp(EntityManagerFactoryScope scope) {
+		scope.inTransaction(
+				entityManager -> {
+					for ( int i = 0; i < 3; i++ ) {
+						Sample sample = new Sample( i, String.valueOf( i ) );
+						entityManager.persist( sample );
+					}
+				}
+		);
+	}
+
+	@Test
+	void testNamedNativeQuery(EntityManagerFactoryScope scope) {
+		scope.inTransaction( em -> {
+			Long count = em.createNamedQuery( "sample.count", Long.class ).getSingleResult();
+			assertThat( count, is( 3l ) );
```

Comment:
this is minor. But I think given the count is long type, `assertThat( count, is( 31L ) )` seems less confusing to me.